docs/configuration_options.rst

```diff
@@ -96,6 +96,8 @@ Data Conversion
   * Importance: low
 
 ``topic.index.map``
+  This option is now deprecated. A future version may remove it completely. Please use single message transforms to map topic names to index names.
```

Comment:
nit: _A future version will remove it completely_?

Reply:
nit: should we also offer some guidance w.r.t appropriate SMTs?
_Please use single message transforms, such as RegexRouter or others, to map topic names ..._

Reply:
Yeah, we can add the guidance. RegexRouter is the way to go. 

Yes, we should be removing this option at a later, as RegexRouter subsumes all functionality of this configuration. Unless, there is a reason to deprecate something and leave it there.

Reply:
I feel that's a bit contradictory. Ones deprecates an option or a feature to signal that it will permanently be removed in the future for sure. If there's ambiguity regarding removal, then it shouldn't be marked as deprecated option.